enclave-runtime/src/rpc/worker_api_direct.rs

```diff
@@ -347,24 +325,20 @@ fn init_io_handler() -> IoHandler {
 		Ok(Value::String("ok".to_owned()))
 	});
 
-	// returns all rpcs methods
-	let rpc_methods_string: String = convert_vec_to_string(rpc_methods_vec);
-	io.add_sync_method("rpc_methods", move |_: Params| {
-		Ok(Value::String(rpc_methods_string.to_owned()))
-	});
-
 	io
 }
 
+/// this is reduced to the side chain block import RPC interface (i.e. worker-worker communication)
+/// the entire rest of the RPC server is run inside the enclave and does not use this e-call function anymore
 #[no_mangle]
 pub unsafe extern "C" fn call_rpc_methods(
 	request: *const u8,
 	request_len: u32,
 	response: *mut u8,
 	response_len: u32,
 ) -> sgx_status_t {
-	// init
-	let io = init_io_handler();
+	//init
+	let io = side_chain_io_handler();
```

Comment:
This function now has the handler for only the side chain block imports, not the entire RPC call interface as before.

Reply:
Ok, so this means that the sidechain block gossiping does still work the same as before?

Reply:
yes, exactly. At least that was the goal, to keep the side chain functionality untouched.